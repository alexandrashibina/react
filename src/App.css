/* Хэдер */

.header {
  min-height: 64px;
  background-color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 2%;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
}

.header__menu-item {
  background-color: transparent;
  border: none;
  color: grey;
  padding: 15px;
  text-align: center;
  display: inline-block;
  cursor: pointer;
}

.header__menu-item:focus {
  outline: transparent;
  color: #FFC617;
}

.header__menu-item:hover {
  color: #FFC617;
}

/* Белый блок на карте */

.block {
  background-color: #fff;
  position: absolute;
  top: 15%;
  left: 2%;
  padding: 50px;
  max-width: 20%;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12);
}

.block__text-header {
  font-size: 35px;
  font-weight: 500;
  margin-bottom: 30px;
}

.block__text-content {
  margin-bottom: 30px;
}

/* Страница логина */

.login {
  display: flex;
  width: 100%;
  justify-content: space-around;
  align-items: center;
}

/* Кнопки */

.button {
  display: flex;
  align-items: flex-end;
  margin-bottom: 30px;
  background-color: #ffc617;
  min-width: 64px;
  border-radius: 4px;
  padding: 6px 16px;
  border-color: transparent;
  cursor: pointer;
}

.button:focus {
  outline: transparent;
}

.button:hover {
  background-color: grey;
  color: #ffffff;
}


/* Форма */

.form {
  min-height: 64px;
  background-color: white;
  margin: 0;
  border-radius: 4px;
  padding: 2%;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
}

.form__input {
  display: flex;
  flex-direction: column;
}

.input {
  margin-bottom: 20px;
  background-color: transparent;
  border: 0;
  border-bottom: 1px solid grey;
  padding: 15px 5px;
  cursor: pointer;
}

.input:focus {
  outline: transparent;
  color: red;
  border-bottom: 1px solid red;
}


.form__header {
  font-size: 35px;
  font-weight: 500;
  margin-bottom: 30px;
}

.form__content {
  display: flex;
  margin-bottom: 30px;
}

.form__text {
  margin-right: 5px;
}

.form__link {
  background-color: transparent;
  border: transparent;
  color: blue;
  text-decoration: underline;
  cursor: pointer;
}

