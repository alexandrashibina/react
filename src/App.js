import React, { Component } from 'react';
import logo from './logo.svg'
import {Map} from './Map.jsx'
import {Login} from './Login.jsx'
import {Profile} from './Profile.jsx'
import {RegForm} from './RegForm.jsx'
import './App.css';



class App extends Component {
  state = {currentPage: "map"}
  
  PAGES = {
    map: <Map/>, 
    profile: <Profile/>, 
    login: <Login onLogin={() => this.navigateTo("map")} onRegister={() => this.navigateTo("reg")}/>, 
    reg: <RegForm/>, 
  };

  navigateTo = (page) => {
    this.setState({currentPage: page})
  }

  renderHeader = () => {
    return (
      <header className="header" id="header">
        {this.hideHeader}
        <div className="header__logo">
        <img src={logo} className="logo-item" alt="logo" />
        </div>
        <nav className="header__menu">
          <ul>
            <li><button className="header__menu-item" onClick={() => {this.navigateTo("map")}}>Карта</button></li>
            <li><button className="header__menu-item" onClick={() => {this.navigateTo("profile")}}>Профиль</button></li>
            <li><button className="header__menu-item" onClick={() => {this.navigateTo("login")}}>Выйти</button></li>
          </ul>
        </nav>
      </header>
    )
  }

  render () {
   return (
    <div className="wrapper">
      {this.state.currentPage !== "login" && this.renderHeader()}
      <main> 
        <section>
          {this.PAGES[this.state.currentPage]}
        </section>
      </main>
    </div>
    )  
  }
} 

export default App;
