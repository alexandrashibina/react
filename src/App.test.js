import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import ReactDOM from 'react-dom';
import App from './App';
import renderer from "react-test-renderer";
import '@testing-library/jest-dom/extend-expect';


jest.mock("mapbox-gl", () => ({
  Map: jest.fn(() => ({ remove: () => {} })),
}));

describe('App component', () => {
  it("renders correctly", () => {
    const tree = renderer.create(
        <App />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("opens correct pages when clicked on buttons", () => {
    const { getByTestId, getByText } = render(
    <App isLoggedIn />
    );

    expect(getByTestId("header")).toBeInTheDocument()
    
    fireEvent.click(getByText('Карта'))
    expect(getByTestId("map")).toBeInTheDocument()

    fireEvent.click(getByText('Профиль'));
    expect(getByTestId("profile")).toBeInTheDocument()
  });
});