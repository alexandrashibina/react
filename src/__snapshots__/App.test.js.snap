// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<< HEAD
exports[`App component renders correctly 1`] = `
=======
exports[`App renders correctly 1`] = `
>>>>>>> feat/week-3
<div
  className="wrapper"
>
  <div>
    <header
      className="header"
      data-testid="header"
<<<<<<< HEAD
=======
      id="header"
>>>>>>> feat/week-3
    >
      <div
        className="header__logo"
      >
        <img
          alt="logo"
          className="logo-item"
          src="logo.svg"
        />
      </div>
      <nav
        className="header__menu"
      >
        <ul>
          <li>
<<<<<<< HEAD
            <button
              className="header__menu-item"
              onClick={[Function]}
            >
              Карта
            </button>
          </li>
          <li>
            <button
              className="header__menu-item"
              onClick={[Function]}
            >
              Профиль
            </button>
          </li>
          <li>
            <button
              className="header__menu-item"
              onClick={[Function]}
            >
              Выйти
            </button>
=======
            <a
              className="header__menu-item"
              href="/map"
              onClick={[Function]}
            >
              Карта
            </a>
          </li>
          <li>
            <a
              className="header__menu-item"
              href="/profile"
              onClick={[Function]}
            >
              Профиль
            </a>
          </li>
          <li>
            <a
              className="header__menu-item"
              href="/"
              onClick={[Function]}
            >
              Выйти
            </a>
>>>>>>> feat/week-3
          </li>
        </ul>
      </nav>
    </header>
  </div>
  <main>
    <section>
<<<<<<< HEAD
      <div>
        <div
          className="block"
        >
          <div
            className="block__text"
          >
            <div
              className="block__text-header"
            >
              Заполните платежные данные
            </div>
            <div
              className="block__text-content"
            >
              Укажите информацию о банковской карте, чтобы сделать заказ.
            </div>
            <button
              className="button"
              onClick={[Function]}
            >
              Перейти в профиль
            </button>
          </div>
        </div>
        <div
          className="map-wrapper"
        >
          <div
            className="map"
            data-testid="map"
          />
        </div>
      </div>
=======
      <p>
        You are logged in. 
        <a
          href="/profile"
          onClick={[Function]}
        >
          Go to profile
        </a>
      </p>
>>>>>>> feat/week-3
    </section>
  </main>
</div>
`;
